### PR TITLE
[CI] Add DCO CI check

### PR DESCRIPTION
**Pull request by foundrytom**
Part of our requirements around Apache 2 licensing.

Can be seen in-action [here](https://github.com/foundrytom/otio-openassetio/pull/2/checks).

### CODE REVIEW

File: .github/workflows/commit-checks.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+on: [pull_request]
+
+jobs:
+  commits_check_job:
+    runs-on: ubuntu-latest
+    name: Check messages
+    steps:
+    - name: Get PR Commits
+      id: 'get-pr-commits'
+      uses: tim-actions/get-pr-commits@master
```
Comment (feltech) [inline .github/workflows/commit-checks.yml]: [consider][minor] Perhaps worth pinning to a specific commit/tag to help avoid breaking changes and/or supply chain attacks.

**Review by feltech — APPROVED** [review]
LGTM

File: .github/workflows/commit-checks.yml (hunk 1 above)
Comment (foundrytom) [reply .github/workflows/commit-checks.yml]: Excellent point, thanks @feltech, pinned [here](https://github.com/TheFoundryVisionmongers/otio-openassetio/pull/2/commits/53d919cffa4dd528a6f134af9ea49963f06026a8).

**Review by feltech — APPROVED** [review]
LGTM, squash and merge when ready.

**Pull request merged**